chia/wallet/lineage_proof.py

```diff
@@ -1,21 +1,43 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import Any, List, Optional
+from enum import Enum
+from typing import Any, Dict, List, Optional
 
 from chia.types.blockchain_format.program import Program
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.util.ints import uint64
 from chia.util.streamable import Streamable, streamable
 
 
+class LineageProofField(Enum):
+    PARENT_NAME = 1
+    INNER_PUZZLE_HASH = 2
+    AMOUNT = 3
+
+
 @streamable
 @dataclass(frozen=True)
 class LineageProof(Streamable):
     parent_name: Optional[bytes32] = None
     inner_puzzle_hash: Optional[bytes32] = None
     amount: Optional[uint64] = None
 
+    @classmethod
+    def from_program(cls, program: Program, fields: List[LineageProofField]) -> LineageProof:
+        program_values = list(program.as_iter())
```

Comment:
I believe `program` cannot be trusted here. But perhaps I'm wrong.
There's a subtle risk when you turn a CLVM structure into a concrete python list. It can be very very long. it would be safer to test the length of this list by iterating `len(fields)` times and ensure the list ends there.
I thought we had a utility function for this, but looking at `Program` and `SExp`, the best I can find is `SExp.list_len()`, which is a little bit better than converting to a `list`